src/server/src/worker-pool.ts

```diff
@@ -18,9 +18,10 @@ const TASK_TIMEOUT_MS = 90_000;
 // Remaining amount is split amongst each worker for topology data
 // This strategy seems to work for any amount of host memory and targets total memory
 // in use maxing out at around 80%
-const dockerMemLimit = Number(
-  readFileSync("/sys/fs/cgroup/memory/memory.limit_in_bytes", { encoding: "ascii" })
-);
+const dockerMemLimit = existsSync("/sys/fs/cgroup/memory.max")
+  ? Number(readFileSync("/sys/fs/cgroup/memory.max", { encoding: "ascii" }))
+  : Number(readFileSync("/sys/fs/cgroup/memory/memory.max_usage_in_bytes", { encoding: "ascii" }));
```

Comment:
Based on the reading I've done, the `max_usage_in_bytes` will "show max memory usage recorded": https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt, which is not the same as what `limit_in_bytes` returns.

And indeed, the value returned by `/sys/fs/cgroup/memory/memory.max_usage_in_bytes` doesn't match the `mem_limit` set in our `docker-compose.yml`, but reverting back to `limit_in_bytes` does https://github.com/PublicMapping/districtbuilder/blob/57b66247386eeb677206d87b806882e0284ce5d2/docker-compose.yml#L47-L48

Tested locally:
```shell
$ docker-compose run --rm --no-deps --entrypoint /bin/bash server -c "cat /sys/fs/cgroup/memory/memory.max_usage_in_bytes"
9113600
$ docker-compose run --rm --no-deps --entrypoint /bin/bash server -c "cat /sys/fs/cgroup/memory/memory.limit_in_bytes"    
12884901888
```

```suggestion
  : Number(readFileSync("/sys/fs/cgroup/memory/memory.limit_in_bytes", { encoding: "ascii" }));
```

Reply:
You are correct - max_usage_in_bytes is not the correct value.

memory.limit_in_bytes appears to be set to unlimited in the actual AWS run container though.. - regardless that doesn't really matter in staging as the totalmem is the full memory.